join-chat-to-teams-meeting/ContentView.swift

```diff
@@ -0,0 +1,341 @@
+
+struct ContentView: View {
+  @State var message: String = ""
+  @State var meetingLink: String = ""
+  @State var chatThreadId: String = ""
+
+  // Calling state
+  @State var callClient: CallClient?
+  @State var callObserver: CallDelegate?
+  @State var callAgent: CallAgent?
+  @State var call: Call?
+
+  // Chat state
+  @State var chatClient: ChatClient?
+  @State var chatThreadClient: ChatThreadClient?
+  @State var chatMessage: String = ""
+  @State var meetingMessages: [MeetingMessage] = []
+
+  let displayName: String = "Quickstart User"
+
+  var body: some View {
+    NavigationView {
+      Form {
+        Section {
+          TextField("Teams Meeting URL", text: $meetingLink)
+            .onChange(of: self.meetingLink, perform: { value in
+              if let threadIdFromMeetingLink = getThreadId(from: value) {
+                self.chatThreadId = threadIdFromMeetingLink
+              }
+            })
+          TextField("Chat thread ID", text: $chatThreadId)
+        }
+        Section {
+          HStack {
+            Button(action: joinMeeting) {
+              Text("Join Meeting")
+            }.disabled(
+              chatThreadId.isEmpty || callAgent == nil || call != nil
+            )
+            Spacer()
+            Button(action: leaveMeeting) {
+              Text("Leave Meeting")
+            }.disabled(call == nil)
+          }
+          Text(message)
+        }
+        Section {
+          ForEach(meetingMessages, id: \.id) { message in
+            let currentUser: Bool = (message.displayName == self.displayName)
+            let foregroundColor = currentUser ? Color.white : Color.black
+            let background = currentUser ? Color.blue : Color(.systemGray6)
+            let alignment = currentUser ? HorizontalAlignment.trailing : .leading
+            
+            HStack {
+              if currentUser {
+                Spacer()
+              }
+              VStack(alignment: alignment) {
+                Text(message.displayName).font(Font.system(size: 10))
+                Text(message.content)
+                  .frame(maxWidth: 200)
+              }
+
+              .padding(8)
+              .foregroundColor(foregroundColor)
+              .background(background)
+              .cornerRadius(8)
+
+              if !currentUser {
+                Spacer()
+              }
+            }
+          }
+          .frame(maxWidth: .infinity)
+        }
+
+        TextField("Enter your message...", text: $chatMessage)
+        Button(action: sendMessage) {
+          Text("Send Message")
+        }.disabled(chatThreadClient == nil)
+      }
+
+      .navigationBarTitle("Teams Chat Interop")
+    }
+
+    .onAppear {
+      if let threadIdFromMeetingLink = getThreadId(from: self.meetingLink) {
+        self.chatThreadId = threadIdFromMeetingLink
+      }
+      // Authenticate
+      do {
+        let credentials = try CommunicationTokenCredential(token: token)
```

Comment:
Recommendation as a quickstart we should push developers to use `CommunicationTokenCredential(withOptions:)` where they can pass in a refresh token using something like this.
> private func creatTokenRefreshOptions(
        initialToken: String? = nil,
        refreshedToken: String? = nil
    ) -> CommunicationTokenRefreshOptions {
        return CommunicationTokenRefreshOptions(
            initialToken: initialToken,
            refreshProactively: true,
            tokenRefresher: creatTokenRefresher(refreshedToken: refreshedToken)
        )
    }

Reply:
We can have that as an update later.